pkg/aws/client.go

```diff
@@ -0,0 +1,57 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+)
+
+// Client provides the single API client to make operations call to aws services
+type Client struct {
+	ec2 EC2Client
+}
+
+// Clients are a map between aws profile and its aws client
+type Clients map[string]*Client
+
+type ServiceEndpoint struct {
+	ServiceID     string
+	URL           string
+	SigningRegion string
+}
+
+type AwsConfigOpt = config.LoadOptionsFunc
+
+func AwsConfigOptSet(opts ...AwsConfigOpt) AwsConfigOpt {
+	return func(conf *config.LoadOptions) error {
+		for _, opt := range opts {
+			if err := opt(conf); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+}
+
+func LoadConfig(ctx context.Context, opts ...AwsConfigOpt) (aws.Config, error) {
+	optFns := []func(*config.LoadOptions) error{}
+
+	for _, opt := range opts {
+		optFns = append(optFns, opt)
+	}
+
+	cfg, err := config.LoadDefaultConfig(ctx, optFns...)
+	if err != nil {
+		return aws.Config{}, fmt.Errorf("setting aws config: %v", err)
```

Comment:
nit: can we just return `nil, fmt.Errorf("setting aws config: %v", err)`?

Reply:
no it's not a pointer